qualtran/bloqs/mcmt/and_bloq.py

```diff
@@ -249,7 +239,13 @@ def _and_bloq() -> And:
 )
 
 
-def _to_tuple(x: Iterable[Union[int, sympy.Expr]]) -> Sequence[Union[int, sympy.Expr]]:
+def _to_tuple(
+    x: Union[SymbolicInt, Iterable[SymbolicInt]]
+) -> Union[Length, Tuple[SymbolicInt, ...]]:
+    if isinstance(x, sympy.Expr):
+        return Length(x)
```

Comment:
wait, shouldn't we make users pass a `HasLength` object explicitly? This is how we do `Shaped` if I'm not mistaken

Reply:
Okay that's right, I'll change it.